File: careerpilot-backend/src/main/java/com/careerpilot/service/AIService.java
package com.careerpilot.service;

import com.careerpilot.dto.ai.AIRequest;
import com.careerpilot.dto.ai.AIResponse;
import com.careerpilot.dto.ai.PromptRequest;
import com.careerpilot.dto.ai.PromptResponse;

/**
 * Central AI service contract.
 * Every AI endpoint in the application calls this service.
 * The implementation is responsible for:
 *   - Delegating to GeminiHttpClient
 *   - Logging every successful request to AIRequestLog
 *   - Measuring and returning processing time
 */
public interface AIService {

    /**
     * Sends a raw prompt to Gemini and returns the response.
     *
     * @param request  Contains the prompt text (and optional context hint)
     * @param userId   The authenticated user's ID — used for request logging
     * @return         AIResponse with the model's text output
     */
    AIResponse sendPrompt(AIRequest request, Long userId);

    /**
     * Resolves a named template, injects the user content, sends to Gemini.
     *
     * @param request  Contains templateKey, userContent, optional extraParams
     * @param userId   The authenticated user's ID — used for request logging
     * @return         PromptResponse wrapping the AIResponse plus the template key used
     */
    PromptResponse sendTemplatePrompt(PromptRequest request, Long userId);
}

File: careerpilot-frontend/src/services/aiService.js
import axiosInstance from './axiosInstance';

/**
 * AI service — thin wrappers around the /api/ai/* endpoints.
 * Every module that needs Gemini calls one of these functions.
 */
const aiService = {
  /**
   * Free-text prompt test.
   * POST /api/ai/test
   * @param {string} prompt - raw prompt string
   */
  test: (prompt) =>
    axiosInstance.post('/api/ai/test', { prompt }),

  /**
   * Template-based prompt.
   * POST /api/ai/prompt
   * @param {string} templateKey  - e.g. "RESUME_ANALYSIS"
   * @param {string} userContent  - content injected into the template
   * @param {string} extraParams  - optional extra context (can be null)
   */
  sendTemplatePrompt: (templateKey, userContent, extraParams = null) =>
    axiosInstance.post('/api/ai/prompt', { templateKey, userContent, extraParams }),
};

export default aiService;
